stats/kafka/config.go

```diff
@@ -0,0 +1,107 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2016 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package kafka
+
+import (
+	"github.com/kubernetes/helm/pkg/strvals"
+	"github.com/mitchellh/mapstructure"
+	"gopkg.in/guregu/null.v3"
+)
+
+type ConfigFields struct {
+	// Connection.
+	Brokers []null.String `json:"brokers" envconfig:"KAFKA_BROKERS"`
```

Comment:
This is not strictly required to be `[]null.String`, since `[]string` actually has a `nil` value - `nil` :smile:. The issue with the normal config `string`/`int`/`bool`/etc. values is that there was no way to differentiate between an empty `""`/`0`/`false`/etc. value and "there is no set value", "the user hasn't supplied anything", etc/ (a.k.a. `null`/`nil`). But for `[]string` we can have an empty slice for a zero value and `nil` for "no value"

Reply:
ah, OK ... thanks for leveling up my go-fu and I'll get this updated.  :)

Reply:
:+1:  and yeah, there's a lot of benefits to Go's simplicity, but its type system leaves much to be desired...